### PR TITLE
refactor(neon_framework): introduce storage library

### PR DESCRIPTION
**Pull request by Leptopoda**
first cleanup step towards #1538

### CODE REVIEW

File: packages/neon_framework/lib/src/storage/single_value_store.dart
Comment (provokateurin) [inline packages/neon_framework/lib/src/storage/single_value_store.dart:1]: Either the file should be called key_value_storage.dart or the class should be called SingleValueStore.

File: packages/neon_framework/lib/src/storage/request_cache.dart
Comment (provokateurin) [inline packages/neon_framework/lib/src/storage/request_cache.dart:1]: The file and class names should be aligned.

**Review by provokateurin — CHANGES_REQUESTED** [review]
I don't like that often the class and file names don't match (not only the ones I mentioned).

**Comment by Leptopoda** [comment]
I already had these changes locally.
I put it into a separate commit to keep the history sorta readable.

File: packages/neon_framework/lib/src/storage/storage_manager.dart [hunk 1]
```diff
@@ -28,39 +28,55 @@ class NeonStorage {
...
 
   /// Initializes a new `KeyValueStorage`.
-  SingleValueStore singleValueStore(StorageKeys key) => DefaultSingleValueStore(key);
+  SingleValueStore singleValueStore(StorageKeys key) {
+    _checkInitialized();
+
+    return DefaultSingleValueStore(key);
+  }
+
+  void _checkInitialized() {
```
Comment (provokateurin) [inline packages/neon_framework/lib/src/storage/storage_manager.dart:81]: Maybe `_assertInitialized`?

File: packages/neon_framework/lib/src/storage/request_cache.dart [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        await batch.commit();
+      },
+    );
+  }
+
+  Database get _requireDatabase {
+    final database = _database;
+    if (database == null) {
+      throw StateError(
+        'Cache has not been set up yet. Please make sure Cache.init() has been called before and completed.',
```
Comment (provokateurin) [inline packages/neon_framework/lib/src/storage/request_cache.dart:66]:
```suggestion
        'Cache has not been set up yet. Please make sure DefaultRequestCache.init() has been called before and completed.',
```

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Review by provokateurin — APPROVED** [review]

**Pull request merged**